knowledge_graphs/query_pagination.py

```
@@ -0,0 +1,382 @@
+from dataclasses import dataclass
+import time
+
+from neo4j import AsyncSession
+
+logger = logging.getLogger(__name__)
+
+@dataclass
+class PaginationConfig:
+    """Configuration for query pagination."""
+    page_size: int = 100
+    max_pages: Optional[int] = None
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)